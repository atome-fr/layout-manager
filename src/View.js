import React from 'react';

/**
 * Method for render the component
 * @param {Object} props The props passed to the component
 */
function View(props) {
    return (
        <div className="component">
            {props.children}
        </div>
    );
}

export default View;