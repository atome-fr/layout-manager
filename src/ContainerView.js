import React, {Component} from 'react';

// Import component
import View from './View';

/**
 * Class used to manage a container view
 * @class
 */
export default class ContainerView extends Component {

    /**
     * Method for render the components
     * @returns {*}
     */
    render() {
        let style = {};
        if (this.props.type === 'row'){
            style['height'] = `calc(${this.props.ratio * 100}% - 5px)`;
        } else{
            style['width'] = `calc(${this.props.ratio * 100}% - 5px)`;
        }
        return (
            <div className={this.props.type} style={style}>
                {
                    React.Children.map(this.props.children, child => {
                        if (child.props.visible && child.type === "div") {
                            return <View>
                                {child}
                            </View>
                        }
                        else if (child.type !== "div") {
                            return child
                        }
                    })
                }
            </div>
        );
    }
}