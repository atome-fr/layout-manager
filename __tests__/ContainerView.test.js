import React from 'react';
import ContainerView from '../src/ContainerView';
import View from '../src/View';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('ContainerView Component', () => {

    describe('UI Render Test', () => {

        it('should render with the ContainerView component in row', () => {
            const wrapper = shallow(<ContainerView type="row" ratio={0.5}>
                <div visible="true"></div>
            </ContainerView>);
            const tree = toJson(wrapper);

            expect(tree).toMatchSnapshot();
            expect(wrapper.find(View).length).toBe(1);
        });
        
        it('should render with the ContainerView component in column', () => {
            const wrapper = shallow(<ContainerView type="column" ratio={0.2}>
                <div visible="true"></div>
            </ContainerView>);
            const tree = toJson(wrapper);

            expect(tree).toMatchSnapshot();
            expect(wrapper.find(View).length).toBe(1);
        });

    });

});