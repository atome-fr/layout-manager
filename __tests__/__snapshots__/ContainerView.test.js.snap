// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerView Component UI Render Test should render with the ContainerView component in column 1`] = `
<div
  className="column"
  style={
    Object {
      "width": "calc(20% - 5px)",
    }
  }
>
  <View
    key=".0"
  >
    <div
      visible="true"
    />
  </View>
</div>
`;

exports[`ContainerView Component UI Render Test should render with the ContainerView component in row 1`] = `
<div
  className="row"
  style={
    Object {
      "height": "calc(50% - 5px)",
    }
  }
>
  <View
    key=".0"
  >
    <div
      visible="true"
    />
  </View>
</div>
`;
